src/modules/keepalive/keepalive_api.c

```diff
@@ -63,9 +64,15 @@ int ka_add_dest(str *uri, str *owner, int flags, ka_statechanged_f callback,
 		void *user_attr)
 {
 	struct sip_uri _uri;
-	ka_dest_t *dest;
+	ka_dest_t *dest=0;
 
-	LM_INFO("adding destination: %.*s\n", uri->len, uri->s);
+	LM_DBG("adding destination: %.*s\n", uri->len, uri->s);
+
+	if(ka_find_destination(uri , owner , &dest , &dest)){
+		LM_INFO("uri [%.*s] already in stack --ignoring \r\n",uri->len, uri->s);
+		dest->counter=0;
```

Comment:
The change of the `dest->counter` is done out of the mutex zone and this operation can be done on an invalid structure if the `dest` was removed meanwhile.